frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector, useDispatch } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import { makeQuery, makeReduxID } from './k8s-watcher';
+import * as k8sActions from '../../actions/k8s';
+import { K8sResourceKind, K8sKind } from '../../module/k8s';
+import { FirehoseResource } from './types';
+import { RootState } from '../../redux';
+
+export class NoModelError extends Error {
+  constructor() {
+    super('Model does not exist');
+  }
+}
+
+export const useK8sWatch = <R extends K8sResourceKind = K8sResourceKind>(
+  resources: FirehoseResource[] = [],
+): WatchK8sResult<R> => {
+  const k8sModelSelectorCreator = React.useMemo(
```

Comment:
Isn't `useCallback` a specialization of `useMemo` for function objects?

If so, shouldn't we use `useCallback` here?

Reply:
no, i want to use useMemo because I want to memoize the result not the call. We dont want to create new selector creator all the time